tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/MongoDb/MongoDbTags.cs

```diff
@@ -31,4 +32,34 @@ internal partial class MongoDbTags : InstrumentationTags
         [Tag(Trace.Tags.OutPort)]
         public string Port { get; set; }
     }
+
+    internal partial class MongoDbV1Tags : MongoDbTags
+    {
+        private string _peerServiceOverride = null;
+
+        // Use a private setter for setting the "peer.service" tag so we avoid
+        // accidentally setting the value ourselves and instead calculate the
+        // value from predefined precursor attributes.
+        // However, this can still be set from ITags.SetTag so the user can
+        // customize the value if they wish.
```

Comment:
Given that the consumer/producer spans are created and closed in non-user code, _can_ they ever set this? 🤔

Reply:
In reality, no 😆 

I was on the fence about implementing this and just making these fields entirely readonly, but we're supposed to allow customization for the "peer.service" tag in general so this implementation is more-or-less a fallback. If it doesn't make sense to keep the setters I'm perfectly fine with removing it!

Reply:
Meh, I'm fine with it either way tbh 😃